index.js

```diff
@@ -102,6 +102,22 @@ module.exports = {
     })
   },
 
+  string ({pattern} = {}) {
+    const s = {type: 'string'}
+    if (pattern) {
+      if (pattern instanceof RegExp) {
+        if (pattern.flags) {
+          throw new Error('JSON schema does not support regexp flags: ' +
+            `${pattern}`)
```

Comment:
I wouldn't break the line like that just because it's barely longer than 80 chars.

Reply:
I'd also only allow the string directly; no regex instance support as that makes the final schema less transparent.
What's your use case here?

Reply:
Using a javascript regex looks much nicer in my editor and even has a visualization. Plus I thought it's nicer if a user doesn't have to think about whether a string or a RegExp needs to be used and both just work. And a string is even more error prone as a user can quickly make mistakes like passing `"/[a-z]/"`.

And except the flags the two are equivalent. So there should be no surprises.